Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -281,21 +459,47 @@ private static void WriteChars(BinaryWriter writer, string str)
             }
         }
 
+        private static string ReadChars(BinaryReader reader)
+        {
+            uint length = reader.ReadUInt32();
+            if (length == 0)
+            {
+                return null;

```

Comment:
>return null; [](start = 16, length = 12)

please add comment to the fact that there is an asymetry now between writing and reading.

`read(write(null)) == null`
`read(write("")) == null `

Ideally you use regular int with  -1 for null or write an extra bit.

Reply:
Yes, I'm aware of this, and will add a comment about that.

---
In reply to: [276813025](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276813025) [](ancestors = 276813025)